Jenkinsfile

```diff
@@ -27,18 +27,51 @@ pipeline {
         script {
           setupGPGEnvironment()
           withCredentials([string(credentialsId: 'gpg.password', variable: 'GPG_PWD')]) {
-            def phase = isReleaseOrMasterBranch() ? 'deploy site-deploy' : 'verify'
+            def phase = isReleaseOrMasterBranch() ? 'deploy' : 'verify'
             maven cmd: "clean ${phase} " +
               "-Dgpg.skip=false " +
               "-Dgpg.project-build.password='${env.GPG_PWD}' " +
-              "-Dgithub.site.skip=${params.skipGitHubSite} " +
               "-Divy.engine.list.url=${params.engineListUrl} " +
               "-Dmaven.test.failure.ignore=true"
           }
           collectBuildArtifacts()
         }
       }
     }
+    stage('deploy-site') {
+      when {
+        expression { !params.skipGitHubSite && isReleaseOrMasterBranch() }
+      }
+      steps {
+        script {
+          maven cmd: "clean site " +
+            "-Divy.engine.list.url=${params.engineListUrl} "
+
+          withEnv(['GIT_SSH_COMMAND=ssh -o StrictHostKeyChecking=no']) {
+            sshagent(credentials: ['github-axonivy']) {
+              def branch = "pages_$BRANCH_NAME-$BUILD_NUMBER"
+              def message = "Update site (${env.BRANCH_NAME})"
+              sh """ 

```

Comment:
the bash script is quite large; can be shared in common .sh script;
as I'm going to use the same code also in the build/release/Jenkinsfile
the common script has the downside of not being editable on re-runs.
do you prefer extracting it anyway?